core/src/main/scala/org/apache/spark/scheduler/SchedulableBuilder.scala

```diff
@@ -69,19 +72,29 @@ private[spark] class FairSchedulableBuilder(val rootPool: Pool, conf: SparkConf)
   val DEFAULT_WEIGHT = 1
 
   override def buildPools() {
-    var is: Option[InputStream] = None
+    var fileData: Option[FileData] = None
     try {
-      is = Option {
-        schedulerAllocFile.map { f =>
-          new FileInputStream(f)
-        }.getOrElse {
-          Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
+      fileData = schedulerAllocFile.map { f =>
+        Some(FileData(new FileInputStream(f), f))
+      }.getOrElse {
+        val is = Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
+        if(is != null) Some(FileData(is, DEFAULT_SCHEDULER_FILE))
+        else {
+          logWarning(s"No Fair Scheduler file found.")
```

Comment:
Can you add the consequence here? (for a user who sees this and wonders why it matters)  I think this would be "Fair Scheduler configuration file not found (so jobs will be scheduled in FIFO order)"